sonnet/src/metrics.py

```diff
@@ -102,8 +102,7 @@ def update(self, value: tf.Tensor):
 
   @property
   def value(self) -> tf.Tensor:
-    """See base class."""
-    # TODO(cjfj): Assert summed type is floating-point?
+    assert type(self.sum) == tf.float32 or type(self.sum) == tf.float64
```

Comment:
That is probably not the check you intended to do. 

`type(self.sum)` returns a Python type of the value in `self.sum`. If `sum` is a a builtin numeric type, this will work (due to the way `tf.DType.__eq__` is implemented), but if `sum` a tensor `type(self.sum)` will be `Tensor` which is incomparable to `tf.DType`.